client/src/App.tsx

```diff
@@ -320,18 +326,24 @@ function App() {
     let _project = (project == 'mathlib-demo' ? null : project)
     if (code === contentFromUrl) {
       if (url !== null) {
-        let args = {project: _project, url: encodeURIComponent(url), code: null}
+        let args = {project: _project, url: encodeURIComponent(url), code: null, codez: null}
         history.replaceState(undefined, undefined!, formatArgs(args))
       } else {
-        let args = {project: _project, url: null, code: null}
+        let args = {project: _project, url: null, code: null, codez: null}
         history.replaceState(undefined, undefined!, formatArgs(args))
       }
     } else if (code === "") {
-      let args = {project: _project, url: null, code: null}
+      let args = {project: _project, url: null, code: null, codez: null}
       history.replaceState(undefined, undefined!, formatArgs(args))
     } else {
-      let args = {project: _project, url: null, code: fixedEncodeURIComponent(code)}
-      history.replaceState(undefined, undefined!, formatArgs(args))
+      const compressed = LZString.compressToBase64(code).replace(/=*$/, '');
```

Comment:
Trimming trailing `=` because (A) it meses with the args parsing (B) it's not necessary anyway (these are padding bytes from base64; but the decompression doesn't actually care if they are there)